File: MojStomatolog/MojStomatolog.Models/Core/Article.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace MojStomatolog.Models.Core
{
    public class Article
    {
        public int ArticleId { get; set; }

        public string Title { get; set; } = string.Empty;

        public string Summary { get; set; } = string.Empty;

        public string Content { get; set; } = string.Empty;

        public DateTime PublishDate { get; set; } = DateTime.Now;

        public int UserCreatedId { get; set; }

        [ForeignKey("UserCreatedId")]
        public User User { get; set; } = null!;
    }
}

File: MojStomatolog/UI/mojstomatolog_desktop/lib/models/article.dart
import 'package:json_annotation/json_annotation.dart';

part 'article.g.dart';

@JsonSerializable()
class Article {
  int? articleId;
  String? title;
  String? summary;
  String? content;
  DateTime? publishDate;
  int? userCreatedId;

  Article();

  factory Article.fromJson(Map<String, dynamic> json) =>
      _$ArticleFromJson(json);
  Map<String, dynamic> toJson() => _$ArticleToJson(this);
}
